File: app/Models/TransactionItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class TransactionItem extends Model
{
    protected $fillable = [
        'transaction_id',
        'product_id',
        'quantity',
        'product_name_snapshot',
        'cost_price_snapshot',
        'selling_price_snapshot',
        'line_subtotal',
    ];

    protected function casts(): array
    {
        return [
            'quantity' => 'integer',
            'cost_price_snapshot' => 'decimal:2',
            'selling_price_snapshot' => 'decimal:2',
            'line_subtotal' => 'decimal:2',
        ];
    }

    public function transaction(): BelongsTo
    {
        return $this->belongsTo(Transaction::class);
    }

    public function product(): BelongsTo
    {
        return $this->belongsTo(Product::class);
    }
}

File: mobile_app/lib/domain/entities/transaction_item.dart
class TransactionItem {
  const TransactionItem({
    required this.id,
    required this.productId,
    required this.isManual,
    required this.productName,
    required this.quantity,
    required this.sellingPrice,
    required this.lineSubtotal,
  });

  final int id;
  final int? productId;
  final bool isManual;
  final String productName;
  final int quantity;
  final double sellingPrice;
  final double lineSubtotal;
}
